packages/notifier/src/notifier.js

```diff
@@ -163,54 +107,51 @@ export const makeNotifierFromAsyncIterable = asyncIterableP => {
 
   /** @type {Promise<UpdateRecord<T>>|undefined} */
   let optNextPromise;
-  /** @type {UpdateCount} */
-  let currentUpdateCount = 1; // avoid falsy numbers
-  /** @type {UpdateRecord<T>|undefined} */
+  /** @type {UpdateCount & bigint} */
+  let currentUpdateCount = 0n;
+  /** @type {ERef<UpdateRecord<T>>|undefined} */
   let currentResponse;
-
-  const hasState = () => currentResponse !== undefined;
-
-  const final = () => currentUpdateCount === undefined;
+  let final = false;
 
   /**
    * @template T
    * @type {BaseNotifier<T>}
    */
   const baseNotifier = Far('baseNotifier', {
-    // NaN matches nothing
-    getUpdateSince(updateCount = NaN) {
-      if (
-        hasState() &&
-        (final() ||
-          (currentResponse && currentResponse.updateCount !== updateCount))
-      ) {
-        // If hasState() and either it is final() or it is
-        // not the state of updateCount, return the current state.
+    getUpdateSince(updateCount = -1n) {
+      if (updateCount < currentUpdateCount) {
+        if (currentResponse) return Promise.resolve(currentResponse);
```

Comment:
Style: we typically avoid one-liner if statements, as they are not valid Jessie.  It's a bug that our lint rules still allow them.

Reply:
yeah, here's the bug https://github.com/Agoric/agoric-sdk/issues/4747